wikimedia/wikipedia-ios
# Fix scroll to section on redirect to section of another article

**Fixes Phabricator ticket:** 
https://phabricator.wikimedia.org/T260429

**Hold pending triage of the ticket**

### Notes
Consolidates article scroll state restoration logic. There are a few different areas we were attempting to scroll to a given area of the page and potentially needed to wait for that scroll offset to become available as the page is loading. This handles those states through a single enum and consolidated logic for retrying after failing to find the offset.

### Test Steps
See ticket for the repro steps of the bug. Also test state restoration (open an article, scroll down the page, force quit the app, re-open) and scroll offset restoration after rotating the device.
